jscomps/LoginManagerPrompter.js

```diff
@@ -90,106 +175,93 @@ LoginManagerPromptFactory.prototype = {
     // already waiting on a master password entry.
     var prompt = this._asyncPrompts[hashKey];
     var prompter = prompt.prompter;
-    var [hostname, httpRealm] = prompter._getAuthTarget(prompt.channel, prompt.authInfo);
-    var hasLogins = (prompter._pwmgr.countLogins(hostname, null, httpRealm) > 0);
-    if (!hasLogins && LoginHelper.schemeUpgrades && hostname.startsWith("https://")) {
-      let httpHostname = hostname.replace(/^https:\/\//, "http://");
-      hasLogins = (prompter._pwmgr.countLogins(httpHostname, null, httpRealm) > 0);
+    var [origin, httpRealm] = prompter._getAuthTarget(
+      prompt.channel,
+      prompt.authInfo
+    );
+    var hasLogins = Services.logins.countLogins(origin, null, httpRealm) > 0;
+    if (
+      !hasLogins &&
+      LoginHelper.schemeUpgrades &&
+      origin.startsWith("https://")
+    ) {
+      let httpOrigin = origin.replace(/^https:\/\//, "http://");
+      hasLogins = Services.logins.countLogins(httpOrigin, null, httpRealm) > 0;
     }
-    if (hasLogins && prompter._pwmgr.uiBusy) {
+    if (hasLogins && Services.logins.uiBusy) {
       this.log("_doAsyncPrompt:run bypassed, master password UI busy");
       return;
     }
 
-    // Allow only a limited number of authentication dialogs when they are all
-    // canceled by the user.
-    var cancelationCounter = (prompter._browser && prompter._browser.canceledAuthenticationPromptCounter) || { count: 0, id: 0 };
-    if (prompt.channel) {
-      var httpChannel = prompt.channel.QueryInterface(Ci.nsIHttpChannel);
-      if (httpChannel) {
-        var windowId = httpChannel.topLevelContentWindowId;
-        if (windowId != cancelationCounter.id) {
-          // window has been reloaded or navigated, reset the counter
-          cancelationCounter = { count: 0, id: windowId };
-        }
-      }
-    }
-
     var self = this;
 
     var runnable = {
       cancel: false,
-      run : function() {
+      run() {
         var ok = false;
         if (!this.cancel) {
           try {
-            self.log("_doAsyncPrompt:run - performing the prompt for '" + hashKey + "'");
-            ok = prompter.promptAuth(prompt.channel,
-                                     prompt.level,
-                                     prompt.authInfo);
+            self.log(
+              "_doAsyncPrompt:run - performing the prompt for '" + hashKey + "'"
+            );
+            ok = prompter.promptAuth(
+              prompt.channel,
+              prompt.level,
+              prompt.authInfo
+            );
           } catch (e) {
-            if (e instanceof Components.Exception &&
-                e.result == Cr.NS_ERROR_NOT_AVAILABLE) {
-              self.log("_doAsyncPrompt:run bypassed, UI is not available in this context");
+            if (
```

Comment:
Have at least first part of the condition on this line.

Reply:
I agree it's odd, but this seems to be the [new general style](https://github.com/sailfishos-mirror/gecko-dev/commit/4a06c925ac8ae7a96aa2d737ceee0508cf8df994#diff-8c9d799cac9363dc2550446f51ea5dad2ca7bbc11128d1dde07827544a1994f8R195) used in upstream gecko components now, so it made things easier [to match](https://github.com/sailfishos-mirror/gecko-dev/blob/2834d64c4b16c7b93857fd58ca55dc76d8176bfd/toolkit/components/passwordmgr/LoginManagerAuthPrompter.jsm#L189) (partly for future changes, but mostly to avoid me having to reformat everything).

If we prefer our own styling, we could tweak mozilla's [Prettier config](https://github.com/sailfishos-mirror/gecko-dev/blob/2834d64c4b16c7b93857fd58ca55dc76d8176bfd/.prettierrc) for our own purposes and run this code through it. What do you think?

Reply:
Then this is perfectly fine. Let's just add reference to the original LoginManagerPrompter.js to the top of the file.